### DISCUSSION TITLE
Reconnect after compile?

### DISCUSSION THREAD

**Discussion by rudolf-adamkovic** [joaotavora/eglot#908] [Ideas]
Actual:

1. Edit some C or Swift code
2. Build the project with `C-x p c`
3. In one of the C or Swift buffers, execute `eglot-reconnect`

Expected:

1. Edit some C or Swift code
2. Build the project with `C-x p c`

In other words, I would like to auto-reconnect (all instances of) Eglot after compilation.

How far I got:

```emacs-lisp
(add-to-list 'compilation-finish-functions
             (lambda (buffer description)
               (eglot-reconnect (eglot-current-server))))
```

This does not work because `eglot-current-server` equals `nil` in the compilation buffer.

Could we have some kind of `eglot-reconnect-all` procedure perhaps?

**Comment by joaotavora** [comment]
Servers are indexed by project and major-mode.  The project should theoretically be determined correctly in the compilation buffer, but the major-mode is probably `compilation-mode`, and thus wrong. You may have some success binding `major-mode` to `c-mode` or `swift-mode` around the call to `eglot-current-server`.  If that works, then maybe a `MODE` parameter can be added to `eglot-current-server`.

**Reply by rudolf-adamkovic** [reply]
@joaotavora I will try that, thank you!  (And I apologize for replying so late.)
